modules/neuwoRtdProvider.js

```diff
@@ -0,0 +1,168 @@
+
+  const wrappedArgUrl = encodeURIComponent(config.params.argUrl || getRefererInfo().page);
+  const url = 'https://m1apidev.neuwo.ai/edge/GetAiTopics?' + [
+    'token=' + config.params.publicToken,
+    'lang=en',
+    'url=' + wrappedArgUrl
+  ].join('&')
+  let billingId = generateUUID();
+  events.emit(CONSTANTS.EVENTS.BILLABLE_EVENT, { type: 'auction', billingId, auctionId: reqBidsConfigObj.auctionId, vendor: neuwoRtdModule.name })
+
+  const success = (responseContent) => {
+    logInfo('NeuwoRTDModule', 'GetAiTopics: response', responseContent)
+    try {
+      var jsonContent = JSON.parse(responseContent);
+      events.emit(CONSTANTS.EVENTS.BILLABLE_EVENT, { type: 'request', billingId, vendor: neuwoRtdModule.name })
+      injectTopics(jsonContent, reqBidsConfigObj, callback, billingId)
+    } catch (ex) {
+      logError('NeuwoRTDModule', 'Response to JSON parse error', ex)
+      callback()
+    }
+  }
+
+  const error = (err) => {
+    logError('xhr error', null, err);
+    callback()
+  }
+
+  ajax(url, {success, error}, null, {
+    // could assume Origin header is set, or
+    // customHeaders: { 'Origin': 'Origin' }
+  })
+}
+
+export function addFragment(base, path, addition) {
+  let container = {}
+  deepSetValue(container, path, addition)
+  mergeDeep(base, container)
+}
+
+/**
+ * Concatenate a base array and an array within an object
+ * non-array bases will be arrays, non-arrays at object key will be discarded
+ * @param {array} base base array to add to
+ * @param {object} source object to get an array from
+ * @param {string} key dot-notated path to array within object
+ * @returns base + source[key] if that's an array
+ */
+function combineArray(base, source, key) {
+  if (Array.isArray(base) === false) base = []
+  let addition = deepAccess(source, key, [])
+  if (Array.isArray(addition)) return base.concat(addition)
+  else return base
+}
+
+export function injectTopics(topics, bidsConfig, callback, billingId) {
+  topics = topics || {}
+
+  // join arrays of IAB category details to single array
+  let combinedTiers = combineArray(
+    combineArray([], topics, RESPONSE_IAB_TIER_1),
+    topics, RESPONSE_IAB_TIER_2)
+
+  let segment = pickSegments(combinedTiers)
+  // effectively gets topics.marketing_categories.iab_tier_1, topics.marketing_categories.iab_tier_2
+  // used as FPD segments content
+
+  let IABSegments = {
+    name: DATA_PROVIDER,
+    ext: { segtax: SEGTAX_IAB },
+    segment
+  }
+
+  if (billingId && segment.length > 0) {
+    events.emit(CONSTANTS.EVENTS.BILLABLE_EVENT, { type: 'general', billingId, vendor: neuwoRtdModule.name })
+  }
+
+  addFragment(bidsConfig.ortb2Fragments.global, 'site.content.data', [IABSegments])
+  logInfo('NeuwoRTDModule', 'injectTopics: post-injection bidsConfig', bidsConfig)
+
+  callback()
+}
+
+/* eslint-disable object-property-newline */
+const D_IAB_ID = { // Content Taxonomy version 2.0 final release November 2017
```

Comment:
you may want to explore which is better for this taxonomy site.content.data as prebid recommends or site.pagecat as https://github.com/InteractiveAdvertisingBureau/openrtb2.x/blob/dea30f23d204e669bd415834b9be096a36eff675/2.6.md?plain=1#L1687 makes available , or both the https://github.com/InteractiveAdvertisingBureau/openrtb/blob/master/extensions/community_extensions/segtax.md

Reply:
category ids now also included in site.pagecat with related change to category taxonomy code field (site.cattax); appreciate the reference(s)